src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,14 +53,84 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
+     *          otherwise prompts the user to select a .csproj file and stores the path
+     *          in the static variable projPath
      */
-    public isDotNetSdkBuild(preLaunchTask: string): boolean {
-        return preLaunchTask === NetSdkTaskFullSymbol;
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+        if (NetSdkDebugHelper.projPath) {
+            return NetSdkDebugHelper.projPath;
+        }
+
+        const projFileItem = await quickPickProjectFileItem(context, undefined, folder, 'No .csproj file could be found.');
+        NetSdkDebugHelper.projPath = projFileItem.absoluteFilePath; // save the path for future use
+        return projFileItem.absoluteFilePath;
+    }
+
+    /**
+     * Overwite the base implementation to infer the container name from the project path (appProject)
+     * instead of the folder name
+     */
+    protected async loadExternalInfo(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<{ configureSsl: boolean, containerName: string, platformOS: PlatformOS }> {
+        NetSdkDebugHelper.projPath = debugConfiguration.netCore.appProject;
+
+        const associatedTask = context.runDefinition;
+        return {
+            configureSsl: !!(associatedTask?.netCore?.configureSsl),
+            containerName: this.inferDotNetSdkContainerName(debugConfiguration),
+            platformOS: associatedTask?.dockerRun?.os || 'Linux',
+        };
+    }
+
+    protected async inferAppOutput(debugConfiguration: DockerDebugConfiguration): Promise<string> {
+        let projectInfo: string[];
+        try {
+            projectInfo = await getNetCoreProjectInfo('GetProjectProperties', debugConfiguration.netCore?.appProject);
+        } catch (error) {
+            await NetSdkDebugHelper.clearWorkspaceState();
+            throw error;
+        }
+
+        if (projectInfo.length >= 5) { // if .NET has support for SDK Build
+            // fifth is whether .NET Web apps supports SDK Containers
+            // sixth is whether .NET Console apps supports SDK Containers
+            if (projectInfo[4] === 'true' || projectInfo[5] === 'true') {
+                return projectInfo[3]; // fourth is output path
+            } else {
+                await NetSdkDebugHelper.clearWorkspaceState();
+                throw new Error(l10n.t("Your current project configuration or .NET SDK version doesn't support SDK Container build. Please choose a compatible project or update .NET SDK."));
+            }
+        }
+
+        throw new Error(l10n.t('Unable to determine assembly output path.'));
+    }
+
+    public static async clearWorkspaceState(): Promise<void> {
+        await ext.context.workspaceState.update(NetContainerBuildOptionsKey, ''); // clear the workspace state
+        NetSdkDebugHelper.projPath = undefined; // clear the static projPath variable
+        return Promise.resolve();
+    }
```

Comment:
Is this still necessary?

Reply:
yeah. this method is only called if there was an error (e.g. if they user's msbuild does not support .NET SDK or dotnet publish fails during execution) so the user can get prompted again to choose which way they want to build (.NET SDK or Dockerfile). 

my thought process was that if an user accidentally chooses the wrong project or action, they should be able to f5 again and chooses something that works (Build with Dockerfile instead of .NET SDK)